spring-integration-core/src/main/java/org/springframework/integration/dispatcher/PartitionedDispatcher.java

```diff
@@ -168,7 +169,7 @@ public boolean dispatch(Message<?> message) {
 		populatedPartitions();
 		int partition = Math.abs(this.partitionKeyFunction.apply(message).hashCode()) % this.partitionCount;
 		UnicastingDispatcher partitionDispatcher = this.partitions.get(partition);
-		return partitionDispatcher.dispatch(message);
+		return Objects.requireNonNull(partitionDispatcher).dispatch(message);
```

Comment:
This is not OK.
The `dispatch()` is a critical runtime execution path, therefore anything extra here would cause a performance degradation.
Right, it could be very minor with this `requireNonNull()`, but I don't want to take my chances.
Let's bite a bullet and mark this method as `@SuppressWarnings("NullAway") // Dataflow analysis limitation`!

Reply:
Understood.   Thank you for catching this!   I'll get an update out later this afternoon!